core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/ImmutablesPropertiesFactory.java

```diff
@@ -0,0 +1,286 @@
+            } catch (IllegalAccessException | NoSuchMethodException e) {
+                throw new IllegalArgumentException("Failed to inspect method " + m, e);
+            }
+        }
+
+        private MethodHandle findBuilderSetter(final Class<?> builderClass, String name, Type type)
+        throws IllegalAccessException, NoSuchMethodException {
+            final List<NoSuchMethodException> failures = new ArrayList<>();
+            final String setName = "set" + name.substring(0, 1).toUpperCase() + name.substring(1);
+            final Set<String> names = new HashSet<>(Arrays.asList(setName, name));
+            for (String tryName : names) {
+                try {
+                    return MethodHandles.lookup().unreflect(builderClass.getMethod(tryName, GenericTypes.getErasedType(type)));
+                } catch (NoSuchMethodException e) {
+                    failures.add(e);
+                }
+            }
+            for (Method m : builderClass.getMethods()) {
+                if (names.contains(m.getName()) && m.getParameterCount() == 1) {
+                    return MethodHandles.lookup().unreflect(m);
+                }
+            }
+            final IllegalArgumentException iae = new IllegalArgumentException("Failed to find builder setter for property " + name + " of " + type);
+            failures.forEach(iae::addSuppressed);
+            throw iae;
+        }
+
+        @Override
+        public PojoBuilder<T> create() {
+            final Object b = builder.get();
+            return new PojoBuilder<T>() {
+                @Override
+                public void set(String property, Object value) {
+                    Unchecked.biFunction(properties.get(property).setter::invoke).apply(b, value);
+                }
+
+                @Override
+                public T build() {
+                    return defn.cast(Unchecked.function(builderBuild::invoke).apply(b));
+                }
+            };
+        }
+    }
+
+    static class ModifiablePojoProperties<T, M extends T> extends BasePojoProperties<T, M> {
+        ModifiablePojoProperties(Type type, Class<T> defn, Class<M> impl, Supplier<M> constructor) {
+            super(type, defn, impl, constructor);
+        }
+
+        @Override
+        protected ImmutablesPojoProperty<T> createProperty(String name, Method m) {
+            final Type propertyType = GenericTypes.resolveType(m.getGenericReturnType(), getType());
+            try {
+                return new ImmutablesPojoProperty<T>(
+                        name,
+                        QualifiedType.of(propertyType).with(Qualifiers.getQualifiers(m)),
+                        m,
+                        isSetMethod(name),
+                        MethodHandles.lookup().unreflect(m),
+                        MethodHandles.lookup().findVirtual(impl, setterName(name), MethodType.methodType(impl, GenericTypes.getErasedType(propertyType))));
```

Comment:
We should test this assumption that modifiables always generate `setX(x)` vs just `x(x)` if the original getter method is just `x()`. The docs only give `setX()` examples but that might just be an omission.

Reply:
It's possible to customize it by changing the annotation `@Value.Style`, however if you do that then we have no clue what the prefix will be (as the annotation is not retained) so I'm not sure how I would actually handle the situation.  Currently it's simply documented to not support custom setter styles.  I could easily craft a builder that does not follow the `setX` pattern but I see it as out of scope to handle that in this first pass, if that's ok, as I simply don't see a clean way to do it without further cooperation from Immutables.

Reply:
Specifically:
```java
@Value.Style:
    /**
     * Modifiable object "setter" method. Used for mutable implementations.
     * Do not confuse with {@link #init()}
     * @return naming template
     */
    String set() default "set*";
```